pkgs/development/compilers/ballerina/default.nix

```diff
@@ -0,0 +1,48 @@
+{ ballerina, lib, writeText, runCommand, makeWrapper, fetchzip, stdenv, jdk11 }:
+let
+  version = "2201.1.0";
+  codeName = "swan-lake";
+  sha256 = "sha256-WZ6CvgnES1indYeMSuC3odDqwR2J27k+D8RktvHsELs=";
+  jdk = jdk11;
```

Comment:
You should instead override `jdk` in `all-packages.nix` like so:
```nix
  ballerina = callPackage ../development/compilers/ballerina {
    jdk = jdk11;
  };

```

Reply:
I see that this would make it easier to override the JDK. Is this what you were aiming for? I had to change the parameter anyway, since `JAVA_HOME` was set improperly (it has to be `${jdk}/lib/openjdk`), and since this way we depend on the JDK being OpenJDK anyway, I’ve changed the function parameter accordingly. PTAL.